File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 

public class Solution
{
	public ListNode ReverseList(ListNode head)
	{
		ListNode current = head;
		ListNode previous = null;
		
		while (current != null)
		{
			ListNode nextNode = current.next;
			
			current.next = previous;
			
			previous = current;
			
			current = nextNode;
		}
		
		return previous;
	}
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy_node = ListNode(0, head)

        fast = dummy_node
        slow = dummy_node

        for _ in range(n):
            fast = fast.next
        
        while fast.next:
            fast = fast.next
            slow = slow.next
        
        slow.next = slow.next.next
        
        return dummy_node.next